x-pack/legacy/plugins/ml/public/data_frame/components/job_create/job_create_form.tsx

```diff
@@ -180,16 +180,20 @@ export const JobCreateForm: SFC<Props> = React.memo(
       }
     };
 
-    if (started === true && progressPercentComplete === undefined) {
+    const isBatchTransform = typeof jobConfig.sync === 'undefined';
+
+    if (started === true && progressPercentComplete === undefined && isBatchTransform) {
       // wrapping in function so we can keep the interval id in local scope
       function startProgressBar() {
         const interval = setInterval(async () => {
           try {
             const stats = await ml.dataFrame.getDataFrameTransformsStats(jobId);
-            const percent = Math.round(stats.transforms[0].state.progress.percent_complete);
-            setProgressPercentComplete(percent);
-            if (percent >= 100) {
-              clearInterval(interval);
+            if (stats && Array.isArray(stats.transforms)) {
+              const percent = Math.round(stats.transforms[0].state.progress.percent_complete);
```

Comment:
does this need a `stats.transforms.length` check too? just in case

Reply:
yes, thanks for spotting! 👍